### PR TITLE
Integration tests of Chiron rotating and regenerating DNS certificates

### PR DESCRIPTION
**Pull request by lei-tang**
- Integration tests of Chiron rotating DNS certificates
- Integration tests of Chiron regenerating DNS certificates

Please provide a description for what this PR is for: https://github.com/istio/istio/issues/17061.

Design documents:

### CODE REVIEW

File: tests/integration/security/chiron/dns_cert_test.go [hunk 1]
```diff
@@ -33,6 +37,48 @@ const (
 	galleyDNSName             = "istio-galley.istio-system.svc"
 	sidecarInjectorSecretName = "dns.istio-sidecar-injector-service-account"
 	sidecarInjectorDNSName    = "istio-sidecar-injector.istio-system.svc"
+
+	caCertUpdated = `-----BEGIN CERTIFICATE-----
```
Comment (yangminzhu) [inline tests/integration/security/chiron/dns_cert_test.go]: Please include the command in the comment to generate these hard-coded cert.

File: tests/integration/security/chiron/dns_cert_test.go [hunk 2]
```diff
@@ -48,5 +94,54 @@ func TestDNSCertificate(t *testing.T) {
...
 			secret.ExamineDNSSecretOrFail(t, sidecarInjectorSecret, sidecarInjectorDNSName)
+
+			// Test certificate regeneration: if a DNS certificate is deleted, Chiron will regenerate it.
+			env := ctx.Environment().(*kube.Environment)
+			istioNs := inst.Settings().IstioNamespace
+			env.DeleteSecret(istioNs, galleySecretName)
+			env.DeleteSecret(istioNs, sidecarInjectorSecretName)
+			// Sleep 10 seconds for the certificate regeneration to take place.
+			t.Log(`sleep 10 seconds for the certificate regeneration to take place ...`)
+			time.Sleep(10 * time.Second)
```
Comment (yangminzhu) [inline tests/integration/security/chiron/dns_cert_test.go]: The following `WaitForSecretToExistOrFail` is already waiting for the secret, why do we need another sleep before it? Can we remove the hard-coded sleep and use the `WaitForSecretToExistOrFail` only?

File: tests/integration/security/chiron/dns_cert_test.go (hunk 1 above)
Comment (lei-tang) [reply tests/integration/security/chiron/dns_cert_test.go]: Done.

File: tests/integration/security/chiron/dns_cert_test.go (hunk 2 above)
Comment (lei-tang) [reply tests/integration/security/chiron/dns_cert_test.go]: WaitForSecretToExistOrFail() uses a watch() to wait for the secret. watch() may fail in some cases due to timing issues (e.g., the secret is deleted but not not yet created), which cause flaky tests. Thus, sleep is used to watch after operations stabilize.

File: tests/integration/security/chiron/dns_cert_test.go [hunk 3]
```diff
@@ -48,5 +101,54 @@ func TestDNSCertificate(t *testing.T) {
 			secret.ExamineDNSSecretOrFail(t, galleySecret, galleyDNSName)
 			t.Log(`checking Sidecar Injector DNS certificate is valid`)
 			secret.ExamineDNSSecretOrFail(t, sidecarInjectorSecret, sidecarInjectorDNSName)
+
+			// Test certificate regeneration: if a DNS certificate is deleted, Chiron will regenerate it.
```
Comment (yangminzhu) [inline tests/integration/security/chiron/dns_cert_test.go]: there are 3 test cases in this function, can we run them separately in `t.Run()` with an explicit name for each test case? This makes it much easier to find out what is failing by just looking at the dashboard without looking further into the log.

Comment (lei-tang) [reply tests/integration/security/chiron/dns_cert_test.go]: Done.

**Pull request merged**